### PR TITLE
Some additions

### PR DESCRIPTION
**Pull request by j0ach1mmall3**
We seriously need Apache though, for ArrayUtils joining (unless someone else has better ideas)
- Yay all materials are in (Source: minecraft wiki)
- Some new Entity interfaces
- Added all the Enchantments and EntityTypes (Source: minecraft wiki)
- Added inventories and ItemMeta

### CODE REVIEW

File: api/src/main/java/com/pocketserver/api/entity/living/hostile/Zombie.java [hunk 1]
```diff
@@ -1,16 +1,19 @@
 package com.pocketserver.api.entity.living.hostile;
 
-import com.pocketserver.api.entity.living.LivingEntity;
+import com.pocketserver.api.entity.living.Ageable;
+import com.pocketserver.api.entity.living.EntityEquipmentHolder;
 
 /**
- * The general Zambie in the server.
```
Comment (TheLightMC) [inline api/src/main/java/com/pocketserver/api/entity/living/hostile/Zombie.java]: HOW DARE YOU.

**Comment by j0ach1mmall3** [comment]
Plz don't kill me I thought it was a typo ;(

**Comment by Fireflies** [comment]
Give your reason behind "We seriously need Apache though" if you'd please?
Streams, Collectors and String provide adequate joining for almost anything. In the case of joining strings, Collectors provides a very flexible implementation.

**Comment by j0ach1mmall3** [comment]
I like railguns.
But fine I'll create a method...

**Comment by Fireflies** [comment]
I'm not sure I understand the need. What is it in ArrayUtils that is so attractive?

**Comment by j0ach1mmall3** [comment]
Joining 2 arrays. Do you have better suggestions?

**Comment by Fireflies** [comment]
Recommending an Apache library just for this is somewhat silly. Any real reason (examples please) we need array joining functionality right now?

**Comment by j0ach1mmall3** [comment]
Enchantment targets (for example Swords and Axes), but there might be a better design to solve it.

**Comment by DarkSeraphim** [comment]
@j0ach1mmall3 something like this?
```public static <T> T[] join(Function<Integer, T[]> arrayFactory, T[] a, T[] b) {
    if (a.length == 0) {
        return b;
    } else if (b.length == 0) {
        return a;
    }
    T[] array = arrayFactory.apply(a.length + b.length);
    System.arraycopy(a, 0, array, 0, a.length);
    System.arraycopy(b, 0, array, a.length, b.length);
    return array;
}``` Usage: `join(Integer[]::new, new Integer[]{0}, new Integer[]{1});`

For primitive types you still need to implement them separately (as it sees int[] as an Object, rather than an Object[])

**Comment by Fireflies** [comment]
Just to be that anal guy, why not use an [IntFunction](https://docs.oracle.com/javase/8/docs/api/java/util/function/IntFunction.html)? ;D
Fixed formatting.

**Comment by j0ach1mmall3** [comment]
I see, interesting method @DarkSeraphim

**Comment by DarkSeraphim** [comment]
@Fireflies generic typed, not targetting ints specifically

**Comment by Fireflies** [comment]
But the function input is always an integer? :P

**Comment by j0ach1mmall3** [comment]
@Fireflies, not really, atm I nedd it for joining Material[]

**Comment by j0ach1mmall3** [comment]
*need

**Comment by DarkSeraphim** [comment]
@j0ach1mmall3 well use the method I provided :smile:

**Comment by j0ach1mmall3** [comment]
Will PR an updated version soon

**Pull request closed without merging**